src/infrastructure/test/base.py

```diff
@@ -0,0 +1,66 @@
+import json
+import os
+
+import pytest
+from sqlalchemy import text
+
+from config import settings
+from infrastructure.database import session_manager
+from infrastructure.test.client import TestClient
+from infrastructure.test.utils import truncate_tables, update_sequence
+
+
+class BaseTest:
+    fixtures: list[str] = []
+    client = TestClient()
+
+    @pytest.fixture(scope="class", autouse=True)
+    async def initialize(self):
+        await truncate_tables()
+        await self.populate_db()
+        await update_sequence()
+
+    async def populate_db(self):
+        for fixture in self.fixtures:
+            await self.load_data(fixture)
+
+    async def load_data(self, relative_path):
+        session = session_manager.get_session()
+        file = open(os.path.join(settings.apps_dir, relative_path), "r")
+        data = json.load(file)
+        for datum in data:
+            columns = ",".join(
+                map(lambda field: f'"{field}"', datum["fields"].keys())
+            )
+            values = ",".join(map(_str_caster, datum["fields"].values()))
+            query = text(
+                f"""
+            insert into "{datum['table']}"({columns}) values ({values})
+            """
+            )
+            await session.execute(query)
+        await session.commit()
```

Comment:
I don't think we actually really need this class.
All of that could be moved to the `conftest.py`

For example any kind of autopopulation may be done by:
```python
@pytest(scope="session", autouse=True)
async def _populate_users(user_create_schema):
     # Check out that I used the user_create_schema that I mentioned in the example above
    await UserCRUD().create(user_create_schema)
```

Reply:
You can create any environment from fixtures which will suit your class tests. if you move creating fixtures to session scope you have to deal with specific fixture data which will affect other tests. Moreover, when you have to create specific database state, it will be hard to create it, because all tests will depend on it. If it is in class scope, each class will have own database independent state.